DQMOffline/Lumi/plugins/ZCounting.cc

```diff
@@ -0,0 +1,465 @@
+      }
+
+    }//End of probe loop over tracks
+
+  }//End of tag loop
+
+}
+
+//
+// -------------------------------------- endLuminosityBlock --------------------------------------------
+//
+void ZCounting::endLuminosityBlock(edm::LuminosityBlock const& lumiSeg, edm::EventSetup const& eSetup)
+{
+  edm::LogInfo("ZCounting") <<  "ZCounting::endLuminosityBlock" << std::endl;
+}
+
+//
+// -------------------------------------- functions --------------------------------------------
+//
+
+void ZCounting::initHLT(const edm::TriggerResults& result, const edm::TriggerNames& triggerNames)
+{
+  for(unsigned int irec=0; irec<fTrigger->fRecords.size(); irec++) {
+    fTrigger->fRecords[irec].hltPathName  = "";
+    fTrigger->fRecords[irec].hltPathIndex = (unsigned int)-1;
+    const std::string pattern = fTrigger->fRecords[irec].hltPattern;
+    if(edm::is_glob(pattern)) {  // handle pattern with wildcards (*,?)
+      std::vector<std::vector<std::string>::const_iterator> matches = edm::regexMatch(triggerNames.triggerNames(), pattern);
+      if(matches.empty()) {
+        edm::LogWarning("ZCounting") << "requested pattern [" << pattern << "] does not match any HLT paths" << std::endl;
+      } else {
+        BOOST_FOREACH(std::vector<std::string>::const_iterator match, matches) {
+          fTrigger->fRecords[irec].hltPathName = *match;
+        }
+      }
+    } else {  // take full HLT path name given
+      fTrigger->fRecords[irec].hltPathName = pattern;
+    }
+    // Retrieve index in trigger menu corresponding to HLT path
+    unsigned int index = triggerNames.triggerIndex(fTrigger->fRecords[irec].hltPathName);
+    if(index < result.size()) {  // check for valid index
+      fTrigger->fRecords[irec].hltPathIndex = index;
+    }
+  }
+}
+
+//--------------------------------------------------------------------------------------------------
+bool ZCounting::isMuonTrigger(ZCountingTrigger::TTrigger triggerMenu, TriggerBits hltBits)
+{
+  return triggerMenu.pass("HLT_IsoMu24_v*",hltBits);
+  //return triggerMenu.pass("HLT_IsoMu27_v*",hltBits);
+}
+
+//--------------------------------------------------------------------------------------------------
+bool ZCounting::isMuonTriggerObj(ZCountingTrigger::TTrigger triggerMenu, TriggerObjects hltMatchBits)
+{
+  return triggerMenu.passObj("HLT_IsoMu24_v*","hltL3crIsoL1sMu22L1f0L2f10QL3f24QL3trkIsoFiltered0p09",hltMatchBits);
+  //return triggerMenu.passObj("HLT_IsoMu27_v*","hltL3crIsoL1sMu22Or25L1f0L2f10QL3f27QL3trkIsoFiltered0p09",hltMatchBits);
+}
+//--------------------------------------------------------------------------------------------------
+bool ZCounting::passMuonID(const reco::Muon& muon, const reco::Vertex& vtx, const std::string idType)
+{//Muon ID selection, using internal function "DataFormats/MuonReco/src/MuonSelectors.cc
+
+  if(idType == "Loose"       && muon::isLooseMuon(muon))      return true;
```

Comment:
please make these strings into enums so that the string comparison happens once per job